File: benchmarks/python/prime.py
#!/usr/bin/env python3

ANSWER = 78_498
MAX_NUMBER_TO_CHECK = 1_000_000

prime_mask = [True] * (MAX_NUMBER_TO_CHECK + 1)
prime_mask[0] = False
prime_mask[1] = False

total_primes_found = 0

for p in range(2, MAX_NUMBER_TO_CHECK + 1):
    if not prime_mask[p]:
        continue

    total_primes_found += 1

    for i in range(2 * p, MAX_NUMBER_TO_CHECK + 1, p):
        prime_mask[i] = False

if total_primes_found != ANSWER:
    raise SystemExit(f"wrong answer: expected {ANSWER}, got {total_primes_found}")

File: benchmarks/node/prime.js
const ANSWER = 78498;
const MAX_NUMBER_TO_CHECK = 1_000_000;

const primeMask = new Array(MAX_NUMBER_TO_CHECK + 1).fill(true);
primeMask[0] = false;
primeMask[1] = false;

let totalPrimesFound = 0;

for (let p = 2; p <= MAX_NUMBER_TO_CHECK; p += 1) {
  if (!primeMask[p]) {
    continue;
  }

  totalPrimesFound += 1;

  for (let i = 2 * p; i <= MAX_NUMBER_TO_CHECK; i += p) {
    primeMask[i] = false;
  }
}

if (totalPrimesFound !== ANSWER) {
  throw new Error(`wrong answer: expected ${ANSWER}, got ${totalPrimesFound}`);
}
